### ISSUE TITLE
[Bug]: Correctly import date adapter locale

### ISSUE THREAD

**Issue by mtdvlpr** [userquin/vuetify-nuxt-module#161]
When using Vuetify directly, the following configuration works to get a DateAdapter with a default locale:
```javascript
import { createVuetify } from 'vuetify'
import { nl } from 'date-fns/locale'
import DateFnsAdapter from '@date-io/date-fns'

export const vuetify = createVuetify({
  date: {
    adapter: new DateFnsAdapter({ locale: nl }),
  },
})
```

When I try to accomplish the same with `vuetify-nuxt-module` I get a 500 error (see [reproduction](https://github.com/mtdvlpr/reproduction/tree/vuetify))

**Comment by userquin** [comment]
I'll need to review date entry, it seems the way to register the date adapter has changed, now we need to use the constructor passing the default locale, I asked in Discord Vuetify server (nuxt channel)

**Comment by userquin** [comment]
the adapter is registered properly, you're using date-dns wrongly, no idea how to use it but the error comes from the `formatByString` call:

![imagen](https://github.com/userquin/vuetify-nuxt-module/assets/6311119/19e1cb90-ab50-4c29-8d58-7a2fa477858f)

**Comment by mtdvlpr** [comment]
It's because of the localize prop. When I use the DateAdapter in Vuetify or directly outside of Vuetify it works. So the locale is not properly handled somewhere in vuetify-nuxt-module

**Comment by userquin** [comment]
locale in Vuetify is `LocaleOptions & RtlOptions` and localize is not there IIRC, luxon is working properly...

it seems we need some conversion, using luxon works, so, Vuetify should initialize it properly or the module?

**Comment by userquin** [comment]
I'm checking the logic in the luxon adapter, but seems date-fns is not working properly, I'll check also the locales from the library, I'll try to add required logic (will take some time since I need to check every adapter)

**Comment by userquin** [comment]
locale is a string in Luxon: https://github.com/dmtrKovalenko/date-io/blob/master/packages/luxon/src/luxon-utils.ts#L36

local can be any: https://github.com/dmtrKovalenko/date-io/blob/master/packages/core/IUtils.d.ts#L73C12-L73C15

Reviewing logic here...

**Comment by mtdvlpr** [comment]
@userquin, I have now also reported the bug at Vuetify, since following the docs for VDatePicker are giving the same error: https://github.com/vuetifyjs/vuetify/issues/18769

**Comment by mtdvlpr** [comment]
So the bug is not yours, but it would still be a nice enhancement to be able to define the locale in the DateAdapter constructor, since that does currently work

**Comment by userquin** [comment]
I'll check this issue tmr, don0t close it

**Comment by mtdvlpr** [comment]
This works in Vuetify:
https://play.vuetifyjs.com/#eNqVVN9v0zAQ/lcsv3SIJh4UCVR1UwcMCR4AwcQL4cFLrq03x7Fsp+s09X/n/KNZlnVT9xLFd999d/fd2X/vqDUlW7fgxOI2v7J0SkWtG+PIHSkNcAd/oo9sycI0NRkl7KhQHVDJMQE1RDDZlFzCQyDhFr+fkbeDV3jIFso+wnvUF2XPKq4dmISeB7ho2C4M8YWCTYgoG2UdSfnJycMOju4KRYhbQQ1TrKWCBW+lu4jnkRTLlRuR7diDYiWI8of740jJUfATUoO1fAnWM2FXMSx+fWFdKI/FTwfNJJYuD3JMO106ru2rQtFxmNCZ1jn2heOZOai1ROCph83WGdc6/IYDKuC4UGCSKRiDVFqU16gi22HZEOwtkWvGejnwaEsjtCMWXKuJ5Gp5UlBnCxrCuuEaWPRXIMyFpJH4CnAeCJlxdXt6pFopsbkZi9RIhH06i+CFWOIaNgo7DRIWtGxqLSSYH9oJJCtoJ25BuZTNzbdgc6aFJCvGrKC83mO/shtvK+hPAxbMGgra+Rw3S3DRff77O2zwv3PWTdVKRD/j/AW2ka2vMcI+tqrCsnu4UO3XoJdQywt7vnGg7K4pX+j9FhUUNfz0TOv35U7ySYjDhQk6xolkNdcDJaPjIQ1mieWunNN2ylhZKQyrQIq1yRU4pnTN5ghjplVO1JBVTT3HnPkHVgnr+uYcbJ1dmuYGtUWSXuvpTh6QKiExw7t8EjMkUyb5pfUp9lKn5+PFGaS4TKFMqAo2eT1kZ2FTTGYA/cZP9DC9BmF9zQaup3XbvXIH5NxB52/zyXF+zF4j6R6mw3UaEiaVBrzdg1w25qBV6uOR+s37x7U+euRfwtsru8fduyHbMQ2jwKsSVoD++w+TrVi4

**Comment by mtdvlpr** [comment]
I've narrowed the bug down to `vuetify-nuxt-module` again (see [reproduction](https://github.com/mtdvlpr/reproduction/tree/vuetify)). The code I'm using there works in [Vuetify playground](https://play.vuetifyjs.com/#eNqVVF1v0zAU/StWXtqJJh4UCVS1UzcYEogvMcQL4cFLbjtvjmPZTtdS9b9zbSdpmo2py0MU33vuvcfHx/m9jYzO6KoCyxeb5NZEk4gXqtSWbEmmgVn4FXJkRxa6LMigxg5S2QKlGBGQfQQVZcYEHAIJM/h+j31beI6LeCHNA7xDfZDmPGfKgq7Rcw/nJW3KEJ9KWPuKrJTGkno+mR3uYLhNJSFhyIT4xX45kGIwCqECjGFLMIhxhHc+HN5uZlvKAq9Jj2fdpZ2DPSbtltteu5NURiMv/rlSCVJG5acWCiUQeOZg01XMlPKffoGbs4xL0HWIkM9uBv8LeUNs6z+urOZySXa7BofFXi3FszsUchUXZQ5ilkYumkaENkNof4qLBBJT2iGHS5NpriwxYCtFBJNLbGdNGvmy9sA1LEZ4KoWqLLLsGARPrQOrDBxYYu+xFmQ3Cp7jCFKbwQXRCUhkKOHedxieoPjdfC0Y+qVmOhyekNlZc9B7YDN41jBGHBr6kFYo0iiMlt2qZFHqgtmLTRg3dKlkxUQFI7wEOfmCD9ngM/D0nEOmNMiMoqJZrEEmC77Ea1pKtIunl0aONBegvynLkWkatQ5NIyZEef/Jx6zGQU08u4Hs7pH4rVm7WBp912BAr9Adbc4yvQQb0pdXX2GN320SHVUJRD+R/AGmFJXjGGAXlcyRdgfn2X70540C/TSXawvSNJtyRPdXMY3QJO+e2Pqe7jgZ+zq8dV7H4Ki4YKqnZEgctsEpge6NtcpMKM1yiWV4f/hKJxIslaqgc4RRXUnLC4jzspjjzOQtzbmx3XACpoivdXmP2mKTztZrxx8xqkbihNfJOEyoQ7Fg18aNeLR1/Xt99gTBr+tSymUO66Tod6feKTrWgHntTvQ4vXplXc16qf/r1tz5I2Y20PmrZHyanNIX2PSRTsfr1G9Yq9Tr2/6eslIfZaUuHlu/fPOQ64Nf3nP6dmh3enduyG4U+aPAq+ItEP35B34Ju0w=).

**Comment by mtdvlpr** [comment]
@userquin, I think this line is the killer:
https://github.com/userquin/vuetify-nuxt-module/blob/1bfbb9eb664cef9f1cc3f87c45b69d6a1425c690/src/vite/vuetify-date-configuration-plugin.ts#L43

The `date-fns` locale objects have functions in them (like the `locale2.month` function that throws the 500 error saying it's not a function). Using JSON.parse(JSON.stringify)) breaks functions in the object. You should import locale explicitly (like adapter) without modifications.

**Comment by mtdvlpr** [comment]
Hi @userquin  you're probably busy, which is fine. Any idea when this issue could be picked up?

**Comment by mtdvlpr** [comment]
@userquin vuetify is working. See my earlier comments.

**Comment by userquin** [comment]
@mtdvlpr rn I'm reviewing this, there are a few erratas in docs, it is `localeMessages`  instead `loadMessages` (I'll fix it), I'm adding a new date-io-playground to the repo to test all date-io adapters:
- :x: @date-io/date-fns (WIP)
- :x: @date-io/moment
- :heavy_check_mark: @date-io/luxon
- :x: @date-io/dayjs
- :x: @date-io/js-joda
- :x: @date-io/date-fns-jalali
- :x: @date-io/jalaali
- :x: @date-io/hijri

**Comment by userquin** [comment]
@mtdvlpr if you want to check it clone this repo and checkout the branch: `nr dev:prepare:date-io && nr dev:date-io`

**Comment by userquin** [comment]
@mtdvlpr should be fixed in `v0.7.3`

**Comment by userquin** [comment]
@mtdvlpr no, but I figured out how to deal with dates, did you check `app.vue` in `date-io-playgorund` folder and the vite plugin?

**Comment by mtdvlpr** [comment]
I will when I get home
